docusaurus/docs/iOS/guides/working-with-the-composer.md

```diff
@@ -0,0 +1,221 @@
+---
+title: Working with the Composer
+---
+
+The `ComposerVC` is the component responsible to create new messages or change existing messages. The composer belongs to the `ChatVC` component that integrates the message list and the composer in one view. But if you are not using our `ChatVC` you can use the composer component in isolation with your own chat view as well. The `ComposerVC` manages all the logic and actions of the `ComposerView`, where the latter is only responsible for the composer's layout.
+
+## Customizing the Composer UI
+By subclassing the `ComposerView` you can customize the layout of the Composer, for example, changing the position of the views, removing views or adding new ones. Since all the logic of the composer lives on the `ComposerVC`, this is the class to override in case the customization depends on data. Here is an example of customizing the composer to look like the iMessage app:
+```swift
+final class iMessageComposerView: ComposerView {
+    override func setUpLayout() {
+        super.setUpLayout()
+
+        leadingContainer.removeArrangedSubview(commandsButton)
+        trailingContainer.removeArrangedSubview(sendButton)
+        inputMessageView.inputTextContainer.addArrangedSubview(sendButton)
+    }
+    
+    override func setUpAppearance() {
+        super.setUpAppearance()
+
+        attachmentButton.setImage(
+            UIImage(systemName: "camera.fill"),
+            for: .normal
+        )
+    }
+}
+
+final class iMessageComposerVC: ComposerVC {
+    override func updateContent() {
+        super.updateContent()
+
+        switch content.state {
+        case .new:
+            composerView.inputMessageView.textView.placeholderLabel.text = "iMessage"
+        default:
+            break
+        }
+    }
+}
+
+// Then, replace the components:
+components.messageComposerVC = iMessageComposerVC.self
+components.messageComposerView = iMessageComposerView.self
+```
+
+**Result:**
+<img src={require("../assets/composer-ui-custom-layout.png").default} width="50%"/>
+
+Since the Composer is built with multiple `ContainerStackView`'s with a similar API of a `UIStackView`, it is very easy to change the Composer layout it is just a question of removing views from containers and adding them back to the desired containers. In this case, we override the `setUpLayout()` lifecycle to remove the `commandsButton` all together and we move the `sendButton` from the `trailingContainer` to the `inputTextContainer`. As a last touch, we change the `attachmentButton` image by overriding the `setUpAppearance()` lifecycle.
```

Comment:
We should show the composer's layout (image) showing the container/element positions here. We mention stacks and components but we don't show where in the composer these stacks/components are located

Reply:
Isn't this layout image to be shown only on the UI Components docs? Because AFAIK; they are populated automatically on the docs right? Should we also add them to the guides, or we can reference the ComposerVC component so they can look at the layout?

Reply:
Since the guide is explaining this stuff, replicating the overview image is necessary, to reduce the amount of jump users need to do. And sometimes, the image in the guide will be more informative (more detailed) than the one in the reference doc, as should be the case here. We mention a lot of elements but don't show them visually